public/app/features/apiserver/server.ts

```diff
@@ -0,0 +1,38 @@
+import { config, getBackendSrv } from '@grafana/runtime';
+
+import { ListOptions, MetaStatus, Resource, ResourceForCreate, ResourceList, ResourceServer } from './types';
+
+export interface GroupVersionResource {
+  group: string;
+  version: string;
+  resource: string;
+}
+
+export class ScopedResourceServer<T = object> implements ResourceServer<T> {
```

Comment:
this should eventually live in `@grafana/runtime` -- but starting here till we see it work in a few projects